papyri/gen.py

```diff
@@ -464,31 +463,17 @@ def gen_main(
     """
     main entry point
     """
-    conffile = Path(target_file).expanduser()
-    if conffile.exists():
-        conf: MutableMapping[str, Any] = toml.loads(conffile.read_text())
-        k0 = next(iter(conf.keys()))
-        config = Config(**conf[k0], dummy_progress=dummy_progress)
-        if exec_ is not None:
-            config.exec = exec_
-        if infer is not None:
-            config.infer = infer
-
-        if len(conf.keys()) != 1:
-            raise ValueError(
-                f"We only support one library at a time for now {conf.keys()}"
-            )
-
-        names = list(conf.keys())
+    target_module_name, conf = load_configuration(target_file)
+    config = Config(**conf, dry_run=dry_run, dummy_progress=dummy_progress)
+    if exec_ is not None:
+        config.exec = exec_
+    if infer is not None:
+        config.infer = infer
 
-    else:
-        sys.exit(f"{conffile!r} does not exists.")
-    assert len(names) == 1, "current we can only do one package at a time"
 
-    tp = os.path.expanduser("~/.papyri/data")
+    target_dir = Path("~/.papyri/data").expanduser()
```

Comment:
This touches things close to what you will need to touch for #74, 

So let me know if you want to me to delay merging this.